### PR TITLE
Update deployment workflow to trigger on push instead of pull_request

### PR DESCRIPTION
**Pull request by binnmti**
## Summary by CodeRabbit

- **New Features**
	- Introduced a Tag-based Deployment workflow for automated code merging and deployment to Firebase.
	- Enhanced deployment process with infrastructure management using Terraform.
	- Notifications for deployment status via Discord webhook.
	- Established comprehensive security rules for Firestore and Firebase Storage.

- **Documentation**
	- Added clear declarations for the new workflows and their jobs in the GitHub Actions configuration.

### CODE REVIEW

File: .github/workflows/tag-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      - name: Configure git
+        run: |
+          git config user.name "GitHub Actions"
+          git config user.email "actions@github.com"
+
+      - name: Merge tag commit to production
+        run: |
+          git checkout production
+          git merge ${{ github.sha }} --no-ff
+          git push origin production
```
[the remark this answers is not in the document]
Comment (binnmti) [reply .github/workflows/tag-deploy.yml]: This code has been deleted

File: .github/workflows/tag-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+          echo "$(cat <<EOF
+          VITE_API_KEY=${{ secrets.PROD_VITE_API_KEY }}
+          VITE_AUTH_DOMAIN=${{ secrets.PROD_VITE_AUTH_DOMAIN }}
+          VITE_PROJECT_ID=${{ secrets.PROD_VITE_PROJECT_ID }}
+          VITE_STORAGE_BUCKET=${{ secrets.PROD_VITE_STORAGE_BUCKET }}
+          VITE_MESSAGING_SENDER_ID=${{ secrets.PROD_VITE_MESSAGING_SENDER_ID }}
+          VITE_APP_ID=${{ secrets.PROD_VITE_APP_ID }}
+          VITE_MEASUREMENT_ID=${{ secrets.PROD_VITE_MEASUREMENT_ID }}
+          EOF
+          )" >> .env.production
```
[the remark this answers is not in the document]
Comment (binnmti) [reply .github/workflows/tag-deploy.yml]: fixed

File: .github/workflows/deploy-production.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
+name: Tag-based Deployment
+
+on:
+  # push:
+  #   branches:
+  #     - production
+  pull_request:
+    branches:
+      - main
```
[the remark this answers is not in the document]
Comment (binnmti) [reply .github/workflows/deploy-production.yml]: fixed

File: .github/workflows/deploy-production.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - name: bun install
+        if: ${{ steps.cache-bun.outputs.cache-hit != 'true' }}
+        run: bun i
+      - name: install firebase-tools
+        run: bun i -g firebase-tools
+      - name: Rewrite .env.production file
+        shell: bash
+        run: |
```
[the remark this answers is not in the document]
Comment (binnmti) [reply .github/workflows/deploy-production.yml:42]: fixed

File: .github/workflows/deploy-production.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
+name: Tag-based Deployment
+
```
[the remark this answers is not in the document]
Comment (binnmti) [reply .github/workflows/deploy-production.yml:2]: fixed

File: firebase-terraform/staging/firestore.rules [hunk 6]
```diff
@@ -0,0 +1,9 @@
+service cloud.firestore { 
+  match /databases/{database}/documents {
+    match /{documents=**} {
+      allow read;
```
[the remark this answers is not in the document]
Comment (binnmti) [reply firebase-terraform/staging/firestore.rules]: I haven't thought about it yet, so I'll skip it for this PR.

**Review by Procrustes5 — APPROVED** [review]
LGTM

**Pull request merged**